generators/app/prompts.js

```diff
@@ -308,10 +328,27 @@ function askForUnitaryEntityOperations(generator, done) {
             when: generator.props.currentEntity,
             type: 'checkbox',
             name: 'currentEntityComponents',
-            validate: input => (_.isEmpty(input) ? 'You have to choose at least one component' : true),
             message: 'Which components would you like to generate?',
             choices: getConcernedComponents(previousConfiguration(generator), generator.props.currentEntity, generator.props.currentPrefix),
-            default: []
+            default: [],
+            validate: input => (_.isEmpty(input) ? 'You have to choose at least one component' : true)
+        },
+        {
+            when: response =>
+                response.currentEntityComponents.includes(constants.CONSUMER_COMPONENT) ||
+                response.currentEntityComponents.includes(constants.PRODUCER_COMPONENT),
+            type: 'list',
+            name: 'topic',
+            message: 'For which topic?',
+            choices: topicsChoices(generator, previousConfiguration(generator)),
+            default: constants.DEFAULT_TOPIC
+        },
+        {
+            when: response => response.topic === constants.CUSTOM_TOPIC,
+            type: 'input',
+            name: 'topicName',
+            message: 'What is the topic name?',
+            validate: input => (_.isEmpty(input) ? 'You have to choose a topic name' : true)
```

Comment:
add a pattern validation to avoid some string issue (like spaces or special character),For example, i can put a topic name like "ok go" and it works, but i think it shouldn't no?
![KafkaTopic](https://user-images.githubusercontent.com/44546269/84128261-a3be2d80-aa40-11ea-8506-cc7e970a95a3.png)

Reply:
You're right!